src/main/java/by/custom/utilcalculator/domain/UserProgress.java

```diff
@@ -31,8 +39,13 @@ public OwnersType getOwnersType() {
         return ownersType;
     }
 
-    public void setOwnersType(final OwnersType ownersType) {
-        cleanStepsAfterCurrent(2);
+    public void setOwnersType(final OwnersType ownersType) throws StepsQueueException {
+        int stepID = 1;
+        cleanStepsAfterCurrent(stepID);
+        stepsData[stepID] = ownersType;
+        if (!passengerStepsChecker.isStepOk(stepsData)) {
```

Comment:
It's better to make validation before all other logic.

Reply:
yes, thanks